src/components/side_panel/side_panel/side_panel.ts

```diff
@@ -204,10 +205,17 @@ export class SidePanel extends Component<Props, SpreadsheetChildEnv> {
   setup() {
     this.state = useState({
       panel: sidePanelRegistry.get(this.props.component),
+      lastSidePanelComponentSeen: this.props.component,
+    });
+    onWillUpdateProps((nextProps) => {
+      this.state.panel = sidePanelRegistry.get(nextProps.component);
+      // changing from one side panel to another should call the onCloseSidePanel as
+      // it may contain important side effects
+      if (this.state.lastSidePanelComponentSeen !== nextProps.component) {
+        this.state.lastSidePanelComponentSeen = nextProps.component;
+        this.props.panelProps?.onCloseSidePanel?.();
+      }
```

Comment:
This feel underwhelmingly complex, but a more simple approach would create an infinite cycle of props updating.

Reply:
the simpler approach being:
if (this.props.component && this.props.component !== nextProps.component) {
	// ... 
}

Reply:
What about handling this in the Spreadsheet component?
It's not rocket science but I think it works and it's simple
```diff
diff --git a/src/components/spreadsheet/spreadsheet.ts b/src/components/spreadsheet/spreadsheet.ts
index a2b65a4e..6445fd5f 100644
--- a/src/components/spreadsheet/spreadsheet.ts
+++ b/src/components/spreadsheet/spreadsheet.ts
@@ -258,6 +258,9 @@ export class Spreadsheet extends Component<SpreadsheetProps, SpreadsheetChildEnv
   }
 
   openSidePanel(panel: string, panelProps: any) {
+    if (this.sidePanel.isOpen && panel !== this.sidePanel.component) {
+      this.sidePanel.panelProps?.onCloseSidePanel?.();
+    }
     this.sidePanel.component = panel;
     this.sidePanel.panelProps = panelProps;
     this.sidePanel.isOpen = true;
```

Reply:
Good call it works, so I can remove this ugly stuff